src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -83,8 +81,10 @@ private void startRecording(Description des) {
                     (gc, gc.getBounds(), getFileFormat(mimeType),
                             outputFormatForScreenCapture, null, null, diagnostics);
             this.screenRecorder.start();
+        } catch (HeadlessException e) {
+            logger.warn("Test recorder does not work with Headless mode");
+            this.headless = true;
```

Comment:
why not just set `this.screenRecorder = null;` so you don't need all the extra complexity in the logics below?